### PR TITLE
Additional slippy conversions for path and polygon

### LINKED ISSUE #33: Add support for path and polygon in /slippy
**Is your feature request related to a problem? Please describe.**
If you use slippy with multiple coordinates, it only returns tiles that touch those points. This forces USSs to determine the actual grids in between, depending on their operation.

**Describe the solution you'd like**
Add support for path (multiple points connected, and all tiles touching the path are returned) and support for single polygon (multiple points connected and closed, and all tiles touching the path or insides of the polygon are returned).

**Additional context**
This can be done by the USS, and they can use the examples in the code to host their own conversion if they are concerned about speed or privacy.

Remember, to best support collaboration among USSs, there are two key tenets to the InterUSS Platform:
1. The InterUSS Platform is focused on facilitating communication amongst actively operating USSs.
2. No details about UAS operations are stored or processed on the InterUSS Platform. 

Regardless of the new features and changes brought to the platform via the open source process, these two tenets must be enforced to ensure protection of operator and consumer privacy and ensure the competition between USSs for drone services remains intact.

### PR DESCRIPTION
**Pull request by hi-kevin**
This creates a slippy utility for point, path and polygon to correctly identify the tiles impacted at the specified zoom level, including in cases where a vertex extends over more than two tiles. This fixes #33. 

This also brings forward some bug fixes from TCL4 to support various OAuth in the header and makes unit tests a little more robust.

### CODE REVIEW

File: datanode/src/slippy_util.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+  if len(points[0]) != 2:
+    raise ValueError('Points must be a list of at least two (lat,lon)')
+  last = None
+  result = []
+  for lat, lon in points:
+    if last:
+      latdiff = math.fabs(last[0] - lat)
+      londiff = math.fabs(last[1] - lon)
+      maxdiff = latdiff if latdiff > londiff else londiff
+      splits = int(math.fabs(maxdiff) / DegreesPerTile(zoom)) + 1
```
Comment (boguwei) [inline datanode/src/slippy_util.py]: I think there are corner case(s) which might not be covered by this function if the split is 1? 

Example test case: 
```
ConvertPathToTile(9, [(37.936541030367316, -122.377713074509), (37.69672993401783, -122.10422390269278)])
```

The line should cross through three grids (82, 198), (82, 197), and (81,197), but the conversion only returns two (81, 197), and (82, 198) because the delta step is too large. In fact the coordinates above representing a flight between Richmond and Hayward, the predominant portion of the path is in the tile which is not returned (82, 197).  

image

Comment (hi-kevin) [reply datanode/src/slippy_util.py]: Thanks, I hadn't visualized the corner cutters. Will fix.

File: datanode/src/slippy_util.py [hunk 2]
```diff
@@ -58,15 +94,48 @@ def ConverCSVtoCoordinates(csv):
   return result
 
 
-def ConvertPointToTile(zoom, latitude, longitude):
-  """Calculation from lat/lon to tile at specific zoom."""
+def convert_tile_to_polygon(zoom, xtile, ytile):
+  """Conversion from lat/lon to tile at specific zoom.
```
[the remark this answers is not in the document]
Comment (hi-kevin) [reply datanode/src/slippy_util.py]: Fixed!

File: datanode/src/slippy_util.py [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+
+  Args:
+      zoom: level of zoom in slippy terms (0-20)
+      xtile: x tile in slippy terms
+      ytile: y tile in slippy terms
+  Raises:
+      TypeError if parameters are not ints
+      ValueError if parameters are not in respective bounds
+  Returns:
+      list of points that make up the tile
```
Comment (BenjaminPelletier) [inline datanode/src/slippy_util.py]: nit: list of (lat, lng) points that make up the tile

File: datanode/src/slippy_util.py [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+  """Conversion from lat/lon to tile at specific zoom.
+
+  Args:
+      zoom: level of zoom in slippy terms (0-20)
+      latitude: decimal degrees latitude for the tile
+      longitude: decimal degrees latitude for the tile
+  Raises:
+      TypeError if zoom is not integer or lat/on is not floats
+      ValueError if parameters are not in respective bounds
+  """
```
Comment (BenjaminPelletier) [inline datanode/src/slippy_util.py]: nit: would be nice to note return values/format

File: datanode/src/slippy_util.py (hunk 3 above)
Comment (hi-kevin) [reply datanode/src/slippy_util.py]: Done.

File: datanode/src/slippy_util.py (hunk 4 above)
Comment (hi-kevin) [reply datanode/src/slippy_util.py]: Done.
